src/dsp_tools/utils/xml_validation_models.py

```diff
@@ -0,0 +1,143 @@
+    return res_list
+
+
+def _get_id_prop_encoding_from_one_resource(resource: etree._Element) -> list[TextValueData] | None:
+    if not (children := list(resource.iterchildren(tag="text-prop"))):
+        return None
+    res_id = resource.attrib["id"]
+    return [_get_prop_encoding_from_one_property(res_id, child) for child in children]
+
+
+def _get_prop_encoding_from_one_property(res_id: str, property: etree._Element) -> TextValueData:
+    prop_name = property.attrib["name"]
+    child_attrib = [x.attrib for x in property.iterchildren()]
+    encodings = {x["encoding"] for x in child_attrib}
+    return TextValueData(res_id, prop_name, encodings)
+
+
+def _check_only_one_valid_encoding_used_all_props(text_props: list[TextValueData]) -> list[TextValueData] | None:
+    wrong_props = [x for x in text_props if not _check_only_one_valid_encoding_used_one_prop(x.encoding)]
+    if len(wrong_props) == 0:
+        return None
+    return wrong_props
+
+
+def _check_only_one_valid_encoding_used_one_prop(text_encodings: set[str]) -> bool:
+    if text_encodings == {"xml"} or text_encodings == {"utf8"}:  # noqa: PLR1714
+        return True
+    return False
+
+
+@dataclass
+class InconsistentTextValueEncodings:
+    """
+    This class takes instances that contain the information about resources and the properties that contain
+    invalid encodings.
+
+    An invalid encoding would be a <text-prop> element, that contains
+    <text encoding="utf8">
+    and
+    <text encoding="xml">
+    It is responsible to communicate the problems to the user.
+    """
+
+    problematic_resources: list[TextValueData]
+
+    def execute_problem_protocol(self) -> tuple[str, pd.DataFrame | None]:
+        """
+        This method composes an error message for the user.
+
+        Returns:
+            the error message and a dataframe with the errors if they exceed 50
+        """
+        msg = (
+            "\nSome <text-prop> elements contain <text> elements that use both 'xml' and 'utf8' encoding.\n"
+            "Only one encoding type can be used per <text-prop> element."
+        )
+        df = self._get_problems_as_df()
+        if len(df) > maximum_prints:
+            return msg, df
+        additional_msg = self._make_msg_from_df(df)
+        return msg + grand_separator + additional_msg, None
+
+    def _get_problems_as_df(self) -> pd.DataFrame:
+        df = pd.DataFrame(
+            {
+                "Resource ID": list(x.resource_id for x in self.problematic_resources),
+                "Property Name": list(x.property_name for x in self.problematic_resources),
+            }
+        )
+        return df.sort_values(by=["Resource ID", "Property Name"], ignore_index=True)
+
+    @staticmethod
+    def _make_msg_from_df(df: pd.DataFrame) -> str:
+        groups = df.groupby(by="Resource ID")
+        return medium_separator.join(
+            [InconsistentTextValueEncodings._make_msg_for_one_resource(str(_id), res_df) for _id, res_df in groups]
```

Comment:
it seems unnatural to me that these 2 methods are static. I would expect them both to be normal instance methods, and to call them with self.xy()

Reply:
They are static because they do not need to have access to any class variables at any point. In case of static functions I prefer to declare them so, because one it is correct and for another it makes testing simpler because you need not create a class to call the function.

Reply:
Even if these methods don't access class variables, they access (static) class methods. And for this, I think that static is less elegant than non-static. For example, I think that on line 137, `InconsistentTextValueEncodings._make_msg_for_one_resource()` is slightly more difficult to understand than `self._make_msg_for_one_resource()`.

But perhaps it's a matter of taste, so if you prefer them to be static, it's okay for me.

Reply:
Thanks for the suggestion, I think I prefer Balduin's suggestion.